src/Inkluzitron/Services/TypeReaders/UserTypeReader.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Discord;
+using Discord.Commands;
+using Discord.WebSocket;
+using GrillBot.App.Extensions.Discord;
+using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace GrillBot.App.Infrastructure.TypeReaders
+{
+    public class UserTypeReader : TypeReader
+    {
+        public override async Task<TypeReaderResult> ReadAsync(ICommandContext context, string input, IServiceProvider services)
+        {
+            // Match on caller
+            if (Regex.IsMatch(input.Trim(), "^(me|j[a|á])$", RegexOptions.IgnoreCase))
+                return TypeReaderResult.FromSuccess(context.User);
+
+            // Match on users in DMs by username and discriminator or only username.
+            IUser user;
+            if (context.Guild == null && context.Client is BaseSocketClient client)
+            {
+                // DM's
+                var discIndex = input.LastIndexOf("#");
+                if (discIndex >= 0)
+                {
+                    var username = input.Substring(0, discIndex);
+
+                    if (ushort.TryParse(input[(discIndex + 1)..], out ushort _))
+                        user = await client.FindUserAsync(username, input[(discIndex + 1)..]);
+                    else
+                        user = await client.FindUserAsync(input, null);
+                }
+                else
+                {
+                    user = await client.FindUserAsync(input, null);
+                }
+
+                if (user != null)
+                    return TypeReaderResult.FromSuccess(user);
+            }
+            else if (context.Guild is SocketGuild guild)
+            {
+                // Finds user directly from discord and get guild user from memory.
+                var matches = await guild.SearchUsersAsync(input);

```

Comment:
Funguje to suprově, ale vzhledem k tomu, že to vyhledává jen od začátku, tak to má problém s nickama typu `./crackono.sh`. Představoval bych si, že by to mělo vzít třeba podle `cracko`.

Navrhoval bych proto před tímto vyhledávat podřetězec kdekoliv v nickname/username a pokud to najde víc shod, tak teprve potom zkusit jen od začátku.

Pokud to nechceš dělat, můžu to napsat.

Reply:
Tohle jsem dodělal.

Reply:
Nice, vyzkoušel jsem to a funguje to pěkně